posts/2021-03-18-async-vision-doc.md

```diff
@@ -0,0 +1,72 @@
+
+Writing the "status quo" stories helps us to compensate for the [curse of knowledge][cok]: the folks working on Async Rust tend to be experts in Async Rust. We've gotten used to the [workarounds] required to be productive, and we know the little tips and tricks that can get you out of a jam. The stories help us gauge the cumulative impact all the paper cuts can have on someone still learning their way around. This gives us the data we need to prioritize.
+
+### ...and then tells how we will change it
+
+The ultimate goal of the vision doc, of course, is not just to tell us where we are now, but where we are going and how  we will get there. Once we've made good progress on the status quo stories, the next step will be start brainstorming stories about the ["shiny future"][sf].
+
+The idea is to replay the same scenarios from the "status quo" document, except 2 or 3 years in the future. For example, maybe Grace has access to a debugging tool that is able to diagnose her stuck tasks and tell her what kind of future they are blocked on, so she doesn't have to grep through the logs. Probably she has better ways to do her logging, too, and better documentation to help her get started. Let's be ambitious!
+
+### Involving the whole community
+
+The async vision document provides a forum where the Async Rust community can plan a great overall experience for Async Rust users. Async Rust was intentionally designed not to have a "one size fits all" mindset, and we don't want to change that. Our goal is to build a shared vision for the end-to-end experience while retaining the loosely coupled, exploration-oriented ecosystem we have built.
+
+The [process][htv] we are using to write the vision doc encourages active collaboration and "position sum" thinking. It begins with a [brainstorming period][b] in which we are inviting people to describe their experiences as "status quo" stories; later on, we'll start accepting "shiny future" stories. At the end of the brainstorming period, we will be giving [awards] and then begin work on [harmonizing] the stories into a coherent whole.
+
+During this brainstorming period, we want to focus on getting as many ideas as we can. Having multiple "shiny futures" that address the same problem is a feature, not a bug, as it will let us mix-and-match later to try and find the best overall plan. Comments and questions will be used as a [tool for improving understanding or sharpening proposals.][c] Presenting alternative ideas is done by [writing an alternative narrative][alt].
+
+### Want to help?
+
+If you'd like to help us to write the vision document, we'd love for you to contribute your experiences and vision! **Right now, we're looking for people to write "status quo" stories.** [You'll find instructions and more here!][htvsq]
```

Comment:
I think this page -- not sure if there's a pending PR to update it perhaps? -- might be useful to expand or structure in a clearer way to drive people to where and how to create the PR. https://rust-lang.github.io/wg-async-foundations/vision/how_to_vision/status_quo.html doesn't (at least based on my relatively quick read through) contain a link to "new PR" or perhaps a template or something like that that someone can open in a HackMD or similar and start writing their story. It feels useful to give that guidance. I'm worried that people aren't finding easy links to "what to do next" here.

Reply:
Yes, I agree. I just landed a PR that has more specific instructions, in fact! We now have a template for folks to use and a bit more clarity on the procedural steps. (I'm thinking maybe it'd be good to make it even more explicit.)

Reply:
Since Github doesn't (afaik) have some kind of PR template, Iin the past I've used an "example PR", maybe I'll create one of those.